recipes/sdl_mixer/cmake/conanfile.py

```diff
@@ -0,0 +1,207 @@
+            tc.variables["SDL2MIXER_VORBIS"] = "STB"
+        elif self.options.vorbis == "vorbisfile":
+            tc.variables["SDL2MIXER_VORBIS"] = "VORBISFILE"
+        elif self.options.vorbis == "tremor":
+            tc.variables["SDL2MIXER_VORBIS"] = "TREMOR"
+        else:
+            tc.variables["SDL2MIXER_VORBIS"] = False
+
+        # TODO: add support for dynamic loading of dependencies
+        tc.variables["SDL2MIXER_DEPS_SHARED"] = False
+        tc.variables["SDL2MIXER_FLAC_LIBFLAC_SHARED"] = False
+        tc.variables["SDL2MIXER_MIDI_FLUIDSYNTH_SHARED"] = False
+        tc.variables["SDL2MIXER_MIDI_TIMIDITY_SHARED"] = False
+        tc.variables["SDL2MIXER_MOD_MODPLUG_SHARED"] = False
+        tc.variables["SDL2MIXER_MOD_XMP_SHARED"] = False
+        tc.variables["SDL2MIXER_MP3_MPG123_SHARED"] = False
+        tc.variables["SDL2MIXER_OGG_SHARED"] = False
+        tc.variables["SDL2MIXER_OPUS_SHARED"] = False
+        tc.variables["SDL2MIXER_SNDFILE_SHARED"] = False
+        tc.variables["SDL2MIXER_VORBIS_TREMOR_SHARED"] = False
+        tc.variables["SDL2MIXER_VORBIS_VORBISFILE_SHARED"] = False
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.set_property("flac", "cmake_file_name", "FLAC")
+        deps.set_property("flac", "cmake_target_name", "FLAC")
+        deps.set_property("fluidsynth", "cmake_file_name", "FluidSynth")
+        deps.set_property("fluidsynth", "cmake_target_name", "FluidSynth::FluidSynth")
+        deps.set_property("libxmp", "cmake_file_name", "libxmp")
+        deps.set_property("libxmp", "cmake_target_name", "libxmp::libxmp")
+        deps.set_property("modplug", "cmake_file_name", "modplug")
+        deps.set_property("modplug", "cmake_target_name", "modplug::modplug")
+        deps.set_property("mpg123", "cmake_file_name", "MPG123")
+        deps.set_property("mpg123", "cmake_target_name", "MPG123::mpg123")
+        deps.set_property("opusfile", "cmake_file_name", "opusfile")
+        deps.set_property("opusfile", "cmake_target_name", "opusfile::opusfile")
+        deps.set_property("tremor", "cmake_file_name", "tremor")
+        deps.set_property("tremor", "cmake_target_name", "tremor::tremor")
+        deps.set_property("vorbis", "cmake_file_name", "vorbisfile")
+        deps.set_property("vorbis::vorbisfile", "cmake_target_name", "vorbisfile::vorbisfile")
+
+        deps.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE.txt", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "SDL2_mixer")
```

Comment:
I don't know if it is a good track to follow to try to fix the error but in sdl_ttf: https://github.com/conan-io/conan-center-index/blob/master/recipes/sdl_ttf/all/conanfile.py#L115
```
suffix = "-static" if Version(self.version) >= "2.20.0" and not self.options.shared else ""
self.cpp_info.components["_sdl2_ttf"].set_property("cmake_target_name",f"SDL2_ttf::SDL2_ttf{suffix}")
```
and in the test [CMakeLists.txt](https://github.com/conan-io/conan-center-index/blob/master/recipes/sdl_ttf/all/test_package/CMakeLists.txt#L7C1-L11C8) they use:
```
if(TARGET SDL2_ttf::SDL2_ttf)
    target_link_libraries(${PROJECT_NAME} PRIVATE SDL2_ttf::SDL2_ttf)
else()
    target_link_libraries(${PROJECT_NAME} PRIVATE SDL2_ttf::SDL2_ttf-static)
endif()
```

Reply:
Yes, it should. That's what the targets exported by the `sdl_mixer` project look like as well: https://github.com/libsdl-org/SDL_mixer/blob/release-2.6.3/CMakeLists.txt#L164-L172